OCC-245: Update OC preview version and fix breaking changes

sarahelsaig opened the pull request:
[OCC-245](https://lombiq.atlassian.net/browse/OCC-245)

[OCC-245]: https://lombiq.atlassian.net/browse/OCC-245?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

sarahelsaig committed:
Update OC preview version.

sarahelsaig committed:
Fix breaking changes.

sarahelsaig committed:
Gut obsolete class.

sarahelsaig committed:
Update OC preview version.

sarahelsaig committed:
Remove extension method because now OrchardCore.Abstractions has the same thing.

sarahelsaig committed:
Add NuGet.Config.

sarahelsaig committed:
Revert "Add NuGet.Config."

This reverts commit 17d2374f997919652f7b142b5594cc7268a092e6.

Piedone commented:
FYI after updating to the latest preview packages, you'll need to change `TotalOfContentTypeBuilder` to have this signature for `Build()`, with `ISchema schema,` added (nothing else needed):

```
public void Build(ISchema schema, FieldType contentQuery, ContentTypeDefinition contentTypeDefinition, ContentItemType contentItemType)
```

Same for `ContentItemTypeBuilder` in the Training Demo.

sarahelsaig committed:
Update OC preview version.

sarahelsaig committed:
Update after API change.

sarahelsaig committed:
Merge branch 'task/system-text-json-migration' into issue/OCC-245

sarahelsaig committed:
Merge branch 'task/system-text-json-migration' into issue/OCC-245

sarahelsaig committed:
Copy NuGet.config.

sarahelsaig committed:
Tyy adding package source mapping too.

sarahelsaig committed:
diagnostics

sarahelsaig committed:
diagnostics

sarahelsaig committed:
also restore

sarahelsaig committed:
more testing

sarahelsaig committed:
Revert nuget.config chages.

sarahelsaig committed:
Merge branch 'task/system-text-json-migration' into issue/OCC-245

sarahelsaig committed:
Copy NuGet.config from NEST.

sarahelsaig committed:
Update OC version.

sarahelsaig committed:
Update class after interface change.

sarahelsaig committed:
Update OC version.

sarahelsaig committed:
LocalizedHtmlStringConverter is not needed.

sarahelsaig committed:
Update OC preview version.

sarahelsaig committed:
Update OC preview version.

sarahelsaig committed:
Merge remote-tracking branch 'origin/issue/OCC-245' into issue/OCC-245

# Conflicts:
#	Lombiq.HelpfulLibraries.LinqToDb/Lombiq.HelpfulLibraries.LinqToDb.csproj
#	Lombiq.HelpfulLibraries.OrchardCore/Lombiq.HelpfulLibraries.OrchardCore.csproj

sarahelsaig committed:
Remove LocalizedHtmlStringConverter reference.

sarahelsaig committed:
spelling

sarahelsaig committed:
Remove unneeded test.

sarahelsaig committed:
Switch publish action branch.

sarahelsaig committed:
Disable add-source-link-package because it breaks dotnet restore.

sarahelsaig committed:
Merge branch 'task/system-text-json-migration' into issue/OCC-245

sarahelsaig committed:
Update OC NuGet versions.

sarahelsaig committed:
Update to latest OC preview version.

sarahelsaig committed:
Update OC preview version.

sarahelsaig committed:
Merge remote-tracking branch 'origin/dev' into issue/OCC-245

sarahelsaig committed:
Merge remote-tracking branch 'origin/task/system-text-json-migration' into issue/OCC-245

Piedone committed:
Updating Refit to 7.1.2

Piedone committed:
Merge remote-tracking branch 'origin/issue/OCC-245' into issue/OCORE-191

sarahelsaig committed:
Update CreateModelMaybeAsync to return the viewmodel even if the TryUpdateModelAsync validation fails. Also docs.

See https://github.com/OrchardCMS/OrchardCore.Commerce/pull/454#discussion_r1674644842.

sarahelsaig committed:
Use CreateModelMaybeAsync.

Piedone committed:
Merge remote-tracking branch 'origin/issue/OCC-245' into issue/OCORE-191

Piedone committed:
Merge pull request #275 from Lombiq/issue/OCORE-191

OCORE-191: Updating Refit to 7.1.2

sarahelsaig committed:
Add ConfigureAsync support to InlineStartup.

sarahelsaig committed:
Merge remote-tracking branch 'origin/issue/OCC-245' into issue/OCC-245

sarahelsaig committed:
Make them nullable.

sarahelsaig committed:
Add PrepareShapeTable.

sarahelsaig committed:
Add Cloudsmith badge

sarahelsaig committed:
This is not actually helpful.

sarahelsaig committed:
Update Readme.md

sarahelsaig committed:
Update OC preview versions.

sarahelsaig committed:
Merge branch 'task/system-text-json-migration' into issue/OCC-245

Piedone commented on Lombiq.HelpfulLibraries.AspNetCore/Localization/LocalizedHtmlStringConverter.cs:
Why is this not needed anymore?

sarahelsaig replied:
To quote Sébastien:

> Because nobody should be able to do it.

I have no idea why it's such a taboo and he refused to explain when I asked. I have reworked the one place in OCC where it was needed [to use an auxiliary class](https://github.com/OrchardCMS/OrchardCore.Commerce/pull/454/files#diff-9fb99695b628fe578655bbc3aca2f33bd245999fb5e7e86b8892eedf7a5aca0dR128-R132) instead.

Piedone commented on Lombiq.HelpfulLibraries.OrchardCore/Lombiq.HelpfulLibraries.OrchardCore.csproj:
Deprecate `ConfigureSmtpSettings` since after https://github.com/OrchardCMS/OrchardCore/pull/15254 SMTP settings should be configured differently.

sarahelsaig committed:
Mark ConfigureSmtpSettings obsolete.

sarahelsaig committed:
Add tags-ignore.

sarahelsaig committed:
Try a different way.

sarahelsaig committed:
Use if condition instead.

sarahelsaig committed:
Use issue branch in publish-nuget.yml too.

sarahelsaig committed:
Update UITT version.

sarahelsaig committed:
NuGet.config

sarahelsaig committed:
Obsolete CreateAdHocShape.

sarahelsaig committed:
Try fixing it for real.

sarahelsaig committed:
Fix ScriptModuleResourceFilter finally.

sarahelsaig committed:
Add HttpRequestInfo.

Piedone commented on Lombiq.HelpfulLibraries.OrchardCore/Shapes/ShapeExtensions.cs:
```diff
@@ -60,6 +61,7 @@ public static T As<T>(this IShape shape)
     /// name="shapeTable"/>, then a new descriptor is added with binding that uses <paramref name="displayAsync"/>.
     /// If <paramref name="type"/> is null or empty, a new random unique name is generated.
     /// </summary>
+    [Obsolete($"This no longer works with the {nameof(DefaultShapeTableManager)}. Use {nameof(HtmlShape)} instead.")]
```
Why doesn't this work anymore, isn't this an unintended breaking change that should be reported in the OC issue tracker?

Also, why didn't any of our tests catch this? One of them should.

sarahelsaig replied:
> isn't this an unintended breaking change that should be reported in the OC issue tracker?

It seemed intentional to me. `shapeTable.Descriptors` is now  a `FrozenDictionary`, meaning it's read-only and [this line](https://github.com/Lombiq/Helpful-Libraries/pull/265/files/c5899d3084d7f8103a8666bc3d52f77de288d9bc..5359efa51f40bdc998a7584559c7c2e21a212c12#diff-4e9972ed44cfa1143a8e183b65f2b5679551be32b3ddb3f58624197d841cd6d4R92) throws an exception. What we do here is kind of hacky anyway, adding a custom `IShape` as I did with [`HtmlShape`](https://github.com/Lombiq/Helpful-Libraries/pull/265/files#diff-cea3a22b01c6eaba78ac9a5e3594ff4551c801d8af150e4a0c130629af4f170cR17) is more elegant and less likely to get me ToS'd by the OC folks.

> Also, why didn't any of our tests catch this? One of them should.

We have a custom `IShapeTableManager` implementation `PerTenantShapeTableManager` used by Lombiq.BaseTheme.

Piedone replied:
So this is something that threw, but only when not BaseTheme is used, right? Since `HtmlShape` is only used in `ScriptModuleResourceFilter`, which in turn in our Vue module, I guess this means that the module broke but only with a built-in them (or if otherwise Base Theme is not enabled).

This is quite specific, I wouldn't expect us to guard against it. However, I'm wondering if the OSOCE UI tests should all use Base Theme instead of enabling one of the built-in themes, and have a separate test for Base Theme itself.

sarahelsaig replied:
I don't think that's necessary, if we go down that road, we'll have to randomly enable or disable modules to see which permutation can cause a weird coincidence.

I've updated `PerTenantShapeTableManager` to also use frozen dictionaries. I didn't see any other meaningful change in `DefaultShapeTableManager`.

Piedone replied:
OK!

Piedone commented on .github/workflows/publish-nuget.yml:
```diff
@@ -7,6 +7,7 @@ on:
 
 jobs:
   publish-nuget:
+    if: ${{ ! contains(github.ref_name, '-preview.') }}
```
```suggestion
    if: ${{ !contains(github.ref_name, '-preview.') }}
```

sarahelsaig committed:
Update .github/workflows/publish-nuget.yml

Co-authored-by: Zoltán Lehóczky <zoltan.lehoczky@lombiq.com>

sarahelsaig committed:
Make ShapeTables created by PerTenantShapeTableManager frozen

to match DefaultShapeTableManager where this change was made in PR https://github.com/OrchardCMS/OrchardCore/pull/15651.
